include/polycap.h

```diff
@@ -0,0 +1,125 @@
+#ifndef POLYCAP_H
+#define POLYCAP_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+//Define constants
+#define HC 1.23984193E-7 //h*c [keV*cm]
+#define N_AVOG 6.022098e+23 //Avogadro constant
+#define R0 2.8179403227e-13 //classical electron radius [cm]
+#define EPSILON 1.0e-30
+
+enum _polycap_profile_type {
+	POLYCAP_PROFILE_CONICAL,
+	POLYCAP_PROFILE_PARABOLOIDAL,
+	POLYCAP_PROFILE_ELLIPSOIDAL,
+};
+
+struct _polycap_description;
+struct _polycap_profile;
+struct _polycap_photon;
+struct _polycap_rng; // our rng struct, which will be mapped to either gsl_rng or easy_rng
+
+typedef enum   _polycap_profile_type polycap_profile_type;
+typedef struct _polycap_description  polycap_description;
+typedef struct _polycap_profile      polycap_profile;
+typedef struct _polycap_photon       polycap_photon;
+typedef struct _polycap_rng          polycap_rng;
+
+typedef struct {
+	double x;
+	double y;
+	double z;
+} polycap_vector3;
+
+// get a new profile for a given type with properties
+polycap_profile* polycap_profile_new(
+	enum polycap_profile_type type,
+	double length,
+	double rad_ext[2],
+	double rad_int[2],
+	double focal_dist[2]); // -> def_cap_profile
+
+// get a new profile from Laszlo's ASCII files
+// perhaps it would be a good idea to define a new filetype that would combine these three into a single file? Best to use something like XML for convenience... this could then be polycap_profile_new_from_xml
+polycap_profile* polycap_profile_new_from_file(
+	const char *single_cap_profile_file,
+	const char *central_axis_file,
+	const char *external_shape_file); // read_cap_profile
+
+// free the polycap_profile structure and its associated data
+void polycap_profile_free(polycap_profile *profile);
+
+// load polycap_description from Laszlo's file. This will recursively call the appropriate polycap_profile_new_* routines. Again here a XML variant could be useful...
+polycap_description* polycap_description_new_from_file(const char *filename);
+
+// get a new polycap_description by providing all its properties... perhaps a simpler variant of this function could be defined that would only set the most important parameters and use defaults for the others??
+polycap_description* polycap_description_new(
+	double sig_rough,
+	double sig_wave
+	double corr_length,
+	double d_source,
+	double d_screen,
+	double src_x,
+	double src_y,
+	double src_sigx,
+	double src_sigy,
+	double src_shiftx,
+	double src_shifty,
+	size_t nelem,
+	int iz[],
+	double wi[],
+	double density,
+	polycap_profile *profile);
+
```

Comment:
Wat betreft de incorporatie in MSIM uiteindelijk zijn zaken zoals d_source, d_screen, en de verscheidene src_* parameters overbodig. Die worden immers dan vervangen door eenvoudig weg het coordinaat van waar het foton op het polycapillair invalt, en welke richting het daar heeft.
Dus als deze header bedoelt is puur voor de inclusie met MSIM (van daar de include/ dir?) dan kan dit vermoedelijk nog een pak eenvoudiger gemaakt worden. Als het de bedoeling is dat dit een algemene header is voor het polycap programma op zich, dan dient het te blijven zoals het is natuurlijk.

Reply:
Mijn idee voor de library API is dat deze zodanig geschreven is dat hij voor elk doeleinde kan gebruikt worden, niet enkel XMI-MSIM. Het is geen enkel probleem dat een bepaald programma slechts een deel van de polycap API gebruikt.

Ik denk dat met deze API het mogelijk is vanuit niets een volledige polycap ray-tracing sessie op te starten, zelfs zonder gebruik te maken van de ASCII input-files.

Reply:
Ah, oke :) Sounds good :)